### PR TITLE
starting to write the intro and the RQGIS section

### PR DESCRIPTION
**Pull request by jannes-m**
just to let you know that I started writing on this chapter and to ask if you agree with what I am doing

### CODE REVIEW

File: cli-gui-integration.Rmd [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 
-An important feature of R is it's command-line interface (CLI).
+An important feature of R is it's command-line interface (CLI). To be more specific, R has been designed from the beginning as an interface to other languages [@Chambers.2016]. By contrast, R has never been intended to be a GIS. This in mind, it is really astonishing how much geofunctionality R is providing today. Nevertheless, many, if not most geoalgorithms (despite rgdal, rgeos, sf, sp, etc.) are missing in R. Even worse, R's memory management almost inhibits the complex processing of big spatial data. Of course, it might be possible to overcome these shortcomings by writing new geo-related R packages which in turn are using C++ power in the background (cite Rcpp). However, that would only duplicate already available and already tested functionality, namely GIS functionality (most of which is written in C++ by the way). Hence, instead of duplicating, we merely need to create an interface to already existing software. This chapter will outline how you can access QGIS (**RQGIS**), SAGA (**RSAGA**) and GRASS (**rgrass7**) from within R, alongside good knowledge of the strengths and weaknesses of 'CLIs vs GUIs'.
```
Comment (Robinlovelace) [inline cli-gui-integration.Rmd]: Interesting point that it was designed to be an interface language, to bring together C and FORTRAN as one person put it.

I think the additional content of this introductory paragraph is good and important but would suggest a few changes, to improve clarity, link to the software, use less hyperbole and more established terms (`geofunctionality` is not in my dictionary! and `really astonishing` sounds a little over-the-top). I would reword the paragraph as something like the following (note the use of one sentence per line while we're in this drafting stage of the book's development):

```
An important feature of R is it's command-line interface (CLI).
The language was originally designed as an interface to and extension of other languages, especially C and FORTRAN, to enable access to statistical algorithms in a user-friendly and intuitive read-evaluate-print loop (REPL)  [@Chambers.2016].
R was not intended to be a GIS, making the breadth of R's geospatial functionality astonishing to many who are unaware of its ability to replicate, and in some cases outperform, established GIS software for many operations on spatial data.

Instead of implementing existing GIS algorithms in R, it makes sense to avoid 'reinventing the wheel' by taking advantage of R's ability to interface with other languages (especially C++, which is used for much low-level and high-performance GIS work).
Using compiled code for new geoprocessing functionality (particularly with the help of the excellent **Rcpp** package) could form the basis of new R packages, building on the success of **sf**.
However, there is already a wide range of algorithms that can be accessed via R's interfaces to dedicated GIS software.
It makes sense to understand these before moving to develop your own optimised algorithms.
For this reason this chapter focusses on 'bridges' to the mature GIS products [QGIS](http://qgis.org/) (via the package **RQGIS**), [SAGA](http://www.saga-gis.org/) (**RSAGA**) and [GRASS](https://grass.osgeo.org/) (**rgrass7**) from within R.
By the end of the chapter you should have working knowledge of the functionality such packages open up, and a more nuanced understanding of the  'CLI vs GUI' debate.

```

File: cli-gui-integration.Rmd [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+<!-- sounds like you would like to focus on open-source GIS? There is also the RPyGeo-package integrating R with ArcGIS. Admittedly, it needs some polishing (a student of ours will take care of this) but maybe we can at least mention it -->
```
Comment (Robinlovelace) [inline cli-gui-integration.Rmd]: No harm in mentioning it but yes the focus is 100% on open source software.

File: cli-gui-integration.Rmd [hunk 3]
```diff
@@ -14,10 +16,72 @@ of flexibility and power waiting for you with the command line. Many times
...
+<!-- Idea: bullet points of times when GUI-based systems (particularly QGIS) are advantageous.
+- digitizing (though there is a new package mapedit from Tim Appelhans that allows for some interactive digitizing in R)
+- georeferencing
+- stereoscopic mapping (think of LiDAR and structure from motion stuff)
+- map production (in case you only want to create a beautiful map once. If you have to produce it over and over again, then maybe CLI is the better approach)
+- zooming and dragging on WMS (though this is also possible with mapview and leaflet)
+- -->
 
-This chapter will outline how you can use portable file formats, packages such as **RQGIS**, **RSAGA** and **rgrass7**, alongside good knowledge of the strengths and weaknesses of 'CLIs vs GUIs'.
+## (R)QGIS
```
Comment (Robinlovelace) [inline cli-gui-integration.Rmd]: What you have done in this section is fantastic - thank you, look forward to testing. Very happy with this as-is.

File: cli-gui-integration.Rmd [hunk 4]
```diff
@@ -14,10 +16,72 @@ of flexibility and power waiting for you with the command line. Many times
...
+
+```{r}
+bike_points <- run_qgis(alg, POLYGONS = lnd, POINTS = cycle_hire, FIELD = "no_bikes", 
+                        OUTPUT = "cycle.shp", load_output = TRUE)
+summary(bike_points$no_bikes)
+sum(bike_points$no_bikes > 0)
+```
+
+
+## (R)SAGA
```
Comment (Robinlovelace) [inline cli-gui-integration.Rmd]: Good stuff - look forward to learning this stuff myself.

File: cli-gui-integration.Rmd [hunk 5]
```diff
@@ -14,10 +16,72 @@ of flexibility and power waiting for you with the command line. Many times
...
+"SAGA CMD is a command line interface to execute SAGA modules. Other than SAGA GUI, it provides no functionality for data management and data visualization. It's purpose is to automate tasks by writing scripts or to execute SAGA modules from within an other application like a web server. This section will describe the necessary setup of your system to use SAGA CMD and execute modules" [https://sourceforge.net/p/saga-gis/wiki/Executing%20Modules%20with%20SAGA%20CMD/](https://sourceforge.net/p/saga-gis/wiki/Executing%20Modules%20with%20SAGA%20CMD/).
+
+hint to vignette("RSAGA-landslides", package = "RSAGA").
+
+
+## GRASS through **rgrass7**
```
Comment (Robinlovelace) [inline cli-gui-integration.Rmd]: Great start.

File: cli-gui-integration.Rmd (hunk 1 above)
Comment (jannes-m) [inline cli-gui-integration.Rmd]: Yeah, sounds way better. Thanks for the improvement.

**Comment by Robinlovelace** [comment]
Many thanks for these changes. Ready to merge?

**Comment by Robinlovelace** [comment]
Also if you'd like to change the name of the chapter I'm open for that - I no longer think it's about CLI vs GUI but about GIS specific vs generic (R).

File: refs.bib [hunk 6]
```diff
@@ -66,4 +66,17 @@ @article{obrien_interactive_2016
   pages = {676--683},
   keywords = {census,choropleth,DataShine,interactive,Open data,population},
   file = {Full Text PDF:/.mozilla/firefox/e3lo19yh.default/zotero/storage/5ZX75T6F/O'Brien and Cheshire - 2016 - Interactive mapping for large, open demographic da.pdf:application/pdf;Snapshot:/.mozilla/firefox/e3lo19yh.default/zotero/storage/BP2FF9K7/17445647.2015.html:text/html}
-}
\ No newline at end of file
+}
+
+
+@book{Chambers.2016,
```
Comment (jannes-m) [inline refs.bib]: Thanks for the advice! I have already added a few further references.

**Comment by jannes-m** [comment]
Well, we could merge. But we can also leave it open since I also would like to add the RSAGA- and rgrass7-sections. And then I would have to create a new pull request. But that's up to you. What would you prefer? And yes, we can think about another title for the chapter, could be as simple as "R/GIS integration" or "How to interface GIS from within R" or "How to use open-source GIS from within R". Still, I think it's nice to have the little CLI/GUI-discussion at the beginning of the chapter. Well, now I really have to prepare my teaching...

**Comment by Robinlovelace** [comment]
Ok, best to merge, merge to master, then you can just push to master. Much simpler!

Thanks.
